src/rpc/rawtransaction.cpp

```diff
@@ -99,6 +100,19 @@ void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue& entry)
 
             in.push_back(Pair("value", ValueFromAmount(spend->getIntDenomination())));
             in.push_back(Pair("valueSat", spend->getIntDenomination()));
+        } else if (txin.IsZerocoinRemint()){
+            std::shared_ptr<sigma::CoinRemintToV3>  remint;
```

Comment:
Better to use `std::unique_ptr` instead due to it more lightweight since we don't need the functionalities of `std::shared_ptr`.

Reply:
We don't have make_unique yet, so instead of `remint = std::make_shared<sigma::CoinRemintToV3>(serData);` I would need to have `remint = std::unique_ptr<sigma::CoinRemintToV3>(new sigma::CoinRemintToV3(serData));` which is ways less readable and more error prone. Also, the unique_ptr varian may work slower, as it has 2 memory allocations, instead of one, which make_shared has. More information is available at: https://herbsutter.com/2013/05/29/gotw-89-solution-smart-pointers/

Reply:
I see. You right. I totally forgot about `std::make_unique`. Agree with readability. Thanks.

BTW, actually `unique_ptr` will not allocated additional memory. What it does it just a container to do RAII.